### ISSUE TITLE
NVDA should have also an object dictionary

### ISSUE THREAD

**Issue by nvaccessAuto** [nvaccess/nvda#1343]
**Reported by djblind on 2011-02-01 22:12**
NVDA should give an ability to overide default names of system objects like radio buttons, tree views, checkboxes, its states, and other standard elements. In some languages, original names of these elements, must be expressed in two, or even in extreme cases 3 words, consuming a lot of time in opinion of some experienced users, even if synthesiser speed is set to 100%. Actual, if you want to change this, you must edit a translation file but for normal users, playing with source files isnt easy. I imagine a dialog, containing list of these standard objects like checkboxes, property sheets and possible values of these objects,, and a text field, where user will enter desired name of element, used by NVDA. For example when some checkbox is selected, it will speak "checkbox yes", "checkbox no", even checkbox can be converted to "check", tree view replaced with "christmass tree", edit area with "write" etc. As I remember, this functionality is implemented in jaws.

**Comment by bhavyashah** [comment]
I actually find this a fascinating feature request. I imagine this being somewhat similar in concept to the Punctuation/Symbol Pronunciation dialog but for web elements, control types and other 'objects'. Please add the feature label to this ticket, and prioritize it if possible. @ehollig

**Comment by Qchristensen** [comment]
Just came across this one.  My two thoughts are:

- If there are specific language instances which are particularly verbose or unhelpful, would it be better to raise those with the translators, as they would affect everyone using that language?
- Can you already do what you need with speech dictionaries?  (I just tried and could get NVDA to call a "button" anything else I wanted).

**Comment by josephsl** [comment]
Hi,

I think it would be best to resolve this at the translations level as control role labels are provided by both Microsoft and translators (early on, Korean translators elected to base control role names on Microsoft terminology). Also, note that this issue addresses speech, but also think about braille as well (this is why in braille, we use abbreviations and Unicode braille pattern characters to represent role names and states, respectively).

Thanks.

**Comment by CyrilleB79** [comment]
@josephsl and @Qchristensen it's not a matter of translation. The issue can arise also in English.

Heere is an example:
For efficiency, I want "heading level 1/2/3/etc." be reported as "H1", "H2", "H3". It should remain a personal setting since we cannot expect all NVDA users to understand "H1" for "heading level 1".

Jaws has this feature for more than 10 years and I was using it at the time I was a Jaws user.

@Qchristensen replacing "button" in the dictionary (e.g. by "push") is not satisfactory since it will replace any occurrence of this word and not only the role. E.g. the following sentence:
"Do not press the big red button."
would be read as:
"Do not press the big red push."

The replacement should occur only for speech since in Braille, the main consideration is already to have the shortest symbol and these symbols/abbreviations need to be learned by heart by Braille users. On the contrary for speech, we have two concurrent requirements:
- the expression should be quite short
- the expression should be easily understandable
